### DISCUSSION TITLE
Customer's domain points to my subdomain that's added in Vercel

### DISCUSSION THREAD

**Discussion by codextech** [vercel/community#1375] [Help] [answered]
Hi,
I have a subdomain added in vercel domain "xyz.subdomain.com" where my application is running and attached to this domain. My customer wants to have their domains(any domain) and my "xyz.subdomain.com" Content(website) should show under their domains
Suppose "Customer A" has a domain e.g: example.com
I can provide some DNS records to my customers (vercel DNS) that point to my xyz.subdomain.com

**I want my customer's domain to point to the subdomain (that is added in vercel) and show my subdomain's content without changing the customer domain URL by providing some value to them (so that they can add their domain provider list) like
cname.vercel.com etc.**

Please note: I don't need to rewrite what is explained here:
https://vercel.com/docs/project-configuration#project-configuration/rewrites

**Please Help**

**Comment by amyegan** [comment, marked as answer]
Hi @codextech! If you want more than one domain to point to the same site, you can use DNS records to [add another custom domain](https://vercel.com/guides/how-do-i-add-a-custom-domain-to-my-vercel-project) to the project

**Reply by codextech** [reply]
@amyegan 
But for how many customers i will perform this action? It's possible if I
can provide some DNS CNAME to my customers. So my customers will add new
records as CNAME to their own domain providers.
Just like I added some other site DNS as a CNAME record in my vercel
![image](https://user-images.githubusercontent.com/21341120/194236888-a6da29c2-bb69-435f-9076-a5934c00ef54.png)
and now my subdomain pointing towards their site.

In the same way, I want my site can be loaded under any domain after my
customer will add some CNAME to records in their domain providers, not in
Vercel.

**Reply by amyegan** [reply]
I want to make sure I understand what you're trying to do. Are you trying to have the client domain redirect to your domain, have your domain redirect to the client domain, or have both domains show the same content?

**Reply by codextech** [reply]
I want to make the client domain show my website content that's hosted on vercel
Just mentioned here:
Customer Custom Domain
https://demo.vercel.pub/platforms-starter-kit

**Reply by amyegan** [reply]
Ah, I see. Let's ask @steven-tey for some help. He wrote that Platforms Starter Kit demo and I bet he knows a lot more than I do about it!

In the meantime, you should read through ["How to Build a Multi-Tenant App with Custom Domains Using Next.js"](https://vercel.com/guides/nextjs-multi-tenant-application). I think you'll want to focus on the parts about [configuring rewrites for multi-tenancy](https://vercel.com/guides/nextjs-multi-tenant-application#4.-configure-rewrites-for-multi-tenancy) and [adding custom domains with the Vercel API](https://vercel.com/guides/nextjs-multi-tenant-application#6.-add-custom-domains-with-the-vercel-api)

**Reply by steven-tey** [reply]
Thanks for tagging me Amy!

Hey Tanzeel! You can also refer to this simple example of how you can let your users add domains to your Vercel project programmatically using the Vercel API: https://github.com/vercel/examples/tree/main/solutions/domains-api

If you want a production-ready example, feel free to also refer to the way I set it up in Dub:
- API route: https://github.com/steven-tey/dub/blob/main/pages/api/projects/%5Bslug%5D/domains/%5Bdomain%5D/index.ts
- Helper lib functions: https://github.com/steven-tey/dub/blob/main/lib/domains.ts
